File: En Go - Compilado/SumaDeDigitos.go
package main

import "fmt"

func main() {
  var sumDigit, extNum, numEntero int64

  fmt.Print("Ingrese un numero entero: ")
  fmt.Scanf("%d", &numEntero)

  for numEntero != 0 {
    extNum = numEntero % 10
    numEntero /= 10
    sumDigit += extNum
  }

  fmt.Println("La suma de los digitos es: ", sumDigit)
}

File: En Kotlin - JVM/SumaDeDigitos.kt
fun main(Args: Array<String>) {
    var sumDigit = 0L
    var numEntero: Long
    var extNum: Long

    print("Ingrese un numero entero: ")
    numEntero = readLine()!!.toLong()

    while (numEntero != 0L) {
        extNum = numEntero % 10
        numEntero /= 10
        sumDigit += extNum
    }

    println("La suma de los digitos es: $sumDigit")
}
